src/Avalonia.Controls/MaskedTextBox.cs

```diff
@@ -0,0 +1,394 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using Avalonia.Input;
+using Avalonia.Input.Platform;
+using Avalonia.Interactivity;
+using Avalonia.Styling;
+
+#nullable enable
+
+namespace Avalonia.Controls
+{
+    public class MaskedTextBox : TextBox, IStyleable
+    {
+        public static readonly StyledProperty<bool> AsciiOnlyProperty =
+             AvaloniaProperty.Register<MaskedTextBox, bool>(nameof(AsciiOnly));
+
+        public static readonly StyledProperty<CultureInfo?> CultureProperty =
```

Comment:
Should it be a Direct property?

Reply:
I would say so https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs#L49